addons/udes_stock/models/stock_picking.py

```diff
@@ -1294,17 +1294,22 @@ def _new_picking_for_group(self, group_key, moves, origin=None):
             ('state', 'in', ['assigned', 'confirmed', 'waiting']),
         ])
         if not picking or len(picking) > 1:
-            picking = self.create({
+            values = {
                 'picking_type_id': picking_type.id,
                 'location_id': src_loc.id,
                 'location_dest_id': dest_loc.id,
                 'group_id': group.id,
-                'origin': origin,
-            })
+            }
+            values.update(kwargs)
+
+            picking = self.create(values)
+
         else:
-            # avoid misleading source document
-            if origin != picking.origin:
-                picking.origin = False
+            # avoid misleading values for extra fields
+            for field, value in kwargs.items():
+                current_value = getattr(picking, field, None)
+                if current_value and current_value != value:
+                    setattr(picking, field, False)
```

Comment:
The above is not very clear to me... The extra values are used to create a picking, if none or multiple ones are found, otherwise are used to unset the respective fields of the instance. I can't understand why by simply looking at the workflow. Would a docstring help perhaps?

Reply:
Yes, I will explain.